### PR TITLE
Ameerul /P2PS-1305 access the share ad link

### PR DESCRIPTION
**Pull request by ameerul-deriv**
## Changes:

Please provide a summary of the change.
- Added routing to the shared advert by adding `advert_id` in the url with the advertiser's id
- Updated custom message for buy/sell advert and fixed/floating rates
- removed react-qrcode-logo library, used qrcode-react instead as there were issues when downloading or scanning the qrcode.
- Handled all edge case scenarios for desktop and responsive: (user not logged in, non-verified p2p users, demo account, real non usd, country of residence does not offer p2p, block users, temp barred users, not enough balance sell advert, inactive, paused,deleted, hidden ads)
- Removed margin-top from dialog scss as it was affecting all dialogs
- Handled edge cases for navigating to advertiser page without advert_id
- Fixed console error from advertiser-page-store.js regarding error is null
- updated test case

### Screenshots:

Please provide some screenshots of the change.
## Short demo of navigation:
https://github.com/ameerul-deriv/deriv-app/assets/103412909/4e9c3c13-20ab-4454-b58f-8f52e25bbbf2

Screenshot 2023-08-24 at 1 16 52 PM
Screenshot 2023-08-24 at 4 21 57 PM
Screenshot 2023-08-24 at 4 22 10 PM
Screenshot 2023-08-24 at 4 22 22 PM
Screenshot 2023-08-24 at 4 22 40 PM
Screenshot 2023-08-24 at 4 22 58 PM

### CODE REVIEW

File: packages/cashier/src/components/error-dialog/error-dialog.tsx [hunk 1]
```diff
@@ -107,6 +107,15 @@ const ErrorDialog = observer(({ className, error = {} }: TErrorDialogProps) => {
                     onConfirm: undefined,
                     message: error_message,
                 });
+            } else if (error_code === 'ShareMyAdsError') {
```
Comment (ameerul-deriv) [inline packages/cashier/src/components/error-dialog/error-dialog.tsx:110]: see below as to why i added this error here

File: packages/components/src/components/dialog/dialog.scss [hunk 2]
```diff
@@ -73,7 +73,6 @@
             justify-content: space-between;
             align-items: center;
             margin-bottom: 1rem;
-            margin-top: 1.4rem;
```
Comment (ameerul-deriv) [inline packages/components/src/components/dialog/dialog.scss:76]: i removed this because this was making all the dialogs have an extra margin at the top, initially caused from an old PR: https://github.com/binary-com/deriv-app/pull/3719. Checked the reset password modal again after removing it and matches with design. Same with other error dialogs

Screenshot 2023-08-22 at 3 06 19 PM
Screenshot 2023-08-23 at 11 12 04 AM
Screenshot 2023-08-23 at 11 06 14 AM
Screenshot 2023-08-23 at 11 28 52 AM

File: packages/components/src/components/vertical-tab/vertical-tab.tsx [hunk 3]
```diff
@@ -160,7 +160,7 @@ const VerticalTab = ({
                         is_routed={is_routed}
                         header_title={header_title}
                     />
-                    {is_floating && tab_headers_note && list[curr_tab_index].has_side_note && (
+                    {is_floating && tab_headers_note && list[curr_tab_index]?.has_side_note && (
```
Comment (ameerul-deriv) [inline packages/components/src/components/vertical-tab/vertical-tab.tsx]: Found an error when we switched from real account to demo account from any p2p page as `has_side_note` was "undefined", so added ? here to fix that. `This issue is also in staging and prod`

File: packages/core/src/index.html [hunk 4]
```diff
@@ -175,11 +175,16 @@
...
+            content="Deriv gives everyone an easy way to participate in the financial markets. Trade with as little as $1 USD on major currencies, stocks, indices, and commodities."
+        />
         <meta property="og:title" content="Deriv" />
         <meta property="og:type" content="website" />
-        <meta property="og:image" content="" />
-        <!-- TODO: Use og_image.gif once the new design is ready -->
+        <meta
+            property="og:image"
+            content="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTR3qsyDtQ_1mlX1AoeV963gFP0UZBTcJOcdK0G802Vfg&s"
+        />
```
Comment (ameerul-deriv) [inline packages/core/src/index.html:187]: Preview link was only showing the text `Deriv`. Needed to update it so we can see a proper preview 

Screenshot 2023-08-24 at 1 16 52 PM

File: packages/components/src/components/dialog/dialog.scss (hunk 2 above)
Comment (ameerul-deriv) [reply packages/components/src/components/dialog/dialog.scss:76]: design for reset password modal is here: https://www.figma.com/file/KEWV1ynSuyagoZIFX7nfoV/Release---v1.2.2---Sign-up%2FLog-in?type=design&node-id=744-19196&mode=design&t=wbXMly0Tp3dTp9Lh-0

File: packages/hooks/src/useIsP2PEnabled.ts [hunk 5]
```diff
@@ -25,6 +25,7 @@ const useIsP2PEnabled = () => {
     return {
         ...rest,
         data: is_p2p_enabled,
+        is_p2p_supported_currency,
```
Comment (ameerul-deriv) [inline packages/hooks/src/useIsP2PEnabled.ts:28]: need to return if the user is using a p2p supported currency inside cashier page so we can show the edge case for if the user's currency is not supported. Easier to get the value from here rather than create another hook to return it

File: packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-card/share-my-ads-card.tsx [hunk 6]
```diff
@@ -1,9 +1,10 @@
 import React from 'react';
-import { QRCode } from 'react-qrcode-logo';
+import { QRCodeSVG } from 'qrcode.react';
```
Comment (ameerul-deriv) [inline packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-card/share-my-ads-card.tsx:2]: changed libraries as the previous one cause downloading and scanning issues on responsive. All the lines get messed up

sell_57

File: packages/core/src/index.html (hunk 4 above)
Comment (ameerul-deriv) [reply packages/core/src/index.html:187]: couldn't add the image via relative path as its not supported for og:image, so the image was taken from deriv com

File: packages/p2p/src/stores/advertiser-page-store.js [hunk 7]
```diff
@@ -107,13 +110,15 @@ export default class AdvertiserPageStore extends BaseStore {
...
-                    this.setHasMoreAdvertsToLoad(list.length >= general_store.list_item_limit);
+                if (response) {
+                    if (response.error) {
+                        this.setErrorMessage(response.error);
+                    } else {
+                        const { list } = response.p2p_advert_list;
+
+                        this.setAdverts(list);
+                        this.setHasMoreAdvertsToLoad(list.length >= general_store.list_item_limit);
+                    }
```
Comment (ameerul-deriv) [inline packages/p2p/src/stores/advertiser-page-store.js:118]: this was the error that showed:

Screenshot 2023-08-21 at 11 19 02 AM

File: packages/hooks/src/useIsP2PEnabled.ts (hunk 5 above)
Comment (farhan-nurzi-deriv) [inline packages/hooks/src/useIsP2PEnabled.ts]: There is `useHasP2PSupportedCurrencies` to check if user has account with p2p supported currency. but, if you need to check the current currency this approach is correct 🙏

File: packages/cashier/src/containers/cashier/cashier.tsx [hunk 8]
```diff
@@ -110,8 +111,26 @@ const Cashier = observer(({ history, location, routes: routes_config }: TCashier
...
+                        code: 'ShareMyAdsError',
+                        message:
+                            currency !== 'USD' && is_svg
+                                ? localize('Deriv P2P is currently unavailable in this currency.')
+                                : localize('Deriv P2P is currently unavailable in your country.'),
+                    });
+                }
+            }
         }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (ameerul-deriv) [inline packages/cashier/src/containers/cashier/cashier.tsx:223]: handled all the edge cases here as this already checks if p2p is not enabled and includes the routing check if it is p2p, just added one more if to check if advert_id is within the search params

File: packages/components/src/components/vertical-tab/vertical-tab.tsx (hunk 3 above)
Comment (ameerul-deriv) [reply packages/components/src/components/vertical-tab/vertical-tab.tsx]: https://github.com/ameerul-deriv/deriv-app/assets/103412909/8e168993-87f8-4590-a15c-0704aacf0d94

File: packages/p2p/src/components/modal-manager/modals/error-modal/error-modal.jsx [hunk 9]
```diff
@@ -21,6 +21,7 @@ const ErrorModal = ({
             has_close_icon={has_close_icon}
             is_open={is_modal_open}
             title={error_modal_title}
+            toggleModal={onClose ?? hideModal}
```
Comment (ameerul-deriv) [inline packages/p2p/src/components/modal-manager/modals/error-modal/error-modal.jsx:24]: for some reason the error modals in p2p can have the x icon but then we can't press them since it was missing the toggleModal function here so i decided to add it

File: packages/p2p/src/stores/advertiser-page-store.js [hunk 10]
```diff
@@ -147,6 +152,37 @@ export default class AdvertiserPageStore extends BaseStore {
         this.setIsLoading(false);
     }
 
+    /** This function gets the advert info from the counterparty when user wants to view a shared advert */
+    getAdvertInfo() {
```
Comment (farrah-deriv) [inline packages/p2p/src/stores/advertiser-page-store.js]: Can we create a hook for this?

File: packages/p2p/src/stores/general-store.js [hunk 11]
```diff
@@ -638,6 +641,10 @@ export default class GeneralStore extends BaseStore {
         this.contact_info = contact_info;
     }
 
+    setCounterpartyAdvertId(counterparty_advert_id) {
+        this.counterparty_advert_id = counterparty_advert_id;
+    }
+
```
Comment (farrah-deriv) [inline packages/p2p/src/stores/general-store.js:647]: Can this be handled in the component? Let's not add more logic to stores.

File: packages/p2p/src/stores/advertiser-page-store.js (hunk 10 above)
Comment (ameerul-deriv) [reply packages/p2p/src/stores/advertiser-page-store.js]: hmmm we could, perhaps if we return a function on the action needed to take, either showing the modal or showing the edge cases? would that work?

File: packages/p2p/src/stores/general-store.js (hunk 11 above)
Comment (ameerul-deriv) [reply packages/p2p/src/stores/general-store.js:647]: ideally it would be best, let me try and see if we can move it to the component buttt i added this because i needed to track if there was an advert id, and when the user navigates to the shared ad, the advert_id in search param disappears and reappears, so it kept causing issues when trying to show the advert... hence why i needed this observable... it was a similar issue for the advertiser id and the order id when niloofar and I changed the routing

File: packages/components/src/components/vertical-tab/vertical-tab.tsx (hunk 3 above)
Comment (farrah-deriv) [reply packages/components/src/components/vertical-tab/vertical-tab.tsx]: I believe this was already fixed. Please update your PR with the latest master then check again.

File: packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-modal.tsx [hunk 12]
```diff
@@ -14,42 +14,64 @@ import { observer } from '@deriv/stores';
...
+    const { account_currency, advertiser_details, id, local_currency, rate_display, rate_type, type } = advert;
+    const { id: advertiser_id } = advertiser_details;
+    const { hideModal, is_modal_open } = useModalManagerContext();
 
     const divRef = React.useRef(null);
-    // TODO: replace with proper url when available
-    const advert_url = window.location.href;
-
-    const { hideModal, is_modal_open } = useModalManagerContext();
+    const advert_url = `${window.location.origin}/cashier/p2p/advertiser?id=${advertiser_id}&advert_id=${id}`;
```
Comment (farhan-nurzi-deriv) [inline packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-modal.tsx]: There is a function called `websiteUrl` from shared to get the current host, maybe you can use it. You can ignore this comment 😄 

```suggestion
    const advert_url = `${websiteUrl()}cashier/p2p/advertiser?id=${advertiser_id}&advert_id=${id}`;
```

File: packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-socials/share-my-ads-socials.tsx [hunk 13]
```diff
@@ -1,14 +1,16 @@
 import React from 'react';
+import classNames from 'classnames';
+import { TFunctionDetailedResult } from 'i18next';
```
Comment (farhan-nurzi-deriv) [inline packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-socials/share-my-ads-socials.tsx]: Why do we need this? Isn't localize function always return string? If not, we need to fix the return type in `Components/i18next`

File: packages/p2p/src/stores/advertiser-page-store.js [hunk 14]
```diff
@@ -147,6 +152,37 @@ export default class AdvertiserPageStore extends BaseStore {
...
+                    this.handleTabItemClick(advert_type);
+                    buy_sell_store.setSelectedAdState(p2p_advert_info);
+                    this.loadMoreAdvertiserAdverts({ startIndex: 0 });
+                    general_store.showModal({ key: 'BuySellModal' });
+                } else {
+                    general_store.showModal({
+                        key: 'ErrorModal',
+                        props: {
+                            error_message: localize("It's either deleted or no longer active."),
+                            error_modal_button_text: localize('OK'),
```
Comment (farhan-nurzi-deriv) [inline packages/p2p/src/stores/advertiser-page-store.js]: We dont need to translate button text here, since it will be passed to `<Localize />` component in the modal. If possible, we can put `error_message` in `<Localize />` component as well.

File: packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-socials/share-my-ads-socials.tsx (hunk 13 above)
Comment (ameerul-deriv) [reply packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-socials/share-my-ads-socials.tsx]: yeahhh theres an issue with the p2p Localize and localize components, they dont return it as a string but return it as that type, we should fix it but we should fix it in a separate card...

File: packages/p2p/src/stores/advertiser-page-store.js (hunk 14 above)
Comment (ameerul-deriv) [reply packages/p2p/src/stores/advertiser-page-store.js]: ah yeah 👍

File: packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-socials/share-my-ads-socials.tsx (hunk 13 above)
Comment (farhan-nurzi-deriv) [reply packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-socials/share-my-ads-socials.tsx]: We fix it in refactoring reusable components PR. You can take it from there, so we don't need to refactor this component later 🙇

Comment (ameerul-deriv) [reply packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-socials/share-my-ads-socials.tsx]: ah great, ill take it now and just place it here 👍

File: packages/cashier/src/containers/cashier/cashier.tsx (hunk 8 above)
Comment (nada-deriv) [reply packages/cashier/src/containers/cashier/cashier.tsx:223]: so is_svg is not required in the dependency list?

File: packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-modal.tsx [hunk 15]
```diff
@@ -67,15 +89,15 @@ const ShareMyAdsModal = ({ advert }: TAdvert) => {
...
-            <Modal
-                has_close_icon
-                is_open={is_modal_open}
-                title={localize('Share this ad')}
-                toggleModal={hideModal}
-                width='71rem'
-            >
-                <Modal.Body className='share-my-ads-modal__body'>
+        <Modal
+            has_close_icon
```
Comment (nada-deriv) [inline packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-modal.tsx]: i think has_close_icon is an optional prop with default value true

File: packages/cashier/src/containers/cashier/cashier.tsx (hunk 8 above)
Comment (ameerul-deriv) [reply packages/cashier/src/containers/cashier/cashier.tsx:223]: so for the is_svg check, i needed to only check the scenario if the currency is not USD and within an svg country as i needed to show the first error message in that scenario, if the user is from a non svg country, we prioritise the second message the be shown since they aren't from an svg country

File: packages/components/src/components/dialog/dialog.scss (hunk 2 above)
Comment (farrah-deriv) [inline packages/components/src/components/dialog/dialog.scss:76]: Please add this to QA checklist.

File: packages/p2p/src/stores/general-store.js (hunk 11 above)
Comment (ameerul-deriv) [reply packages/p2p/src/stores/general-store.js:647]: so i tried to handle everything in app.jsx where i used useState to track the advert_id in search params. This did work, but the modal couldn't close because it still had a value within the state. And because that state is only handled in the app, i can't change the value... You can see that i had set `counterparty_advert_id' to an empty string so it wont show the modal. 

i did try to reset the state in app, but then the modal wouldn't show

**Review by adrienne-deriv — APPROVED** [review]
LGTM

**Review by farhan-nurzi-deriv — APPROVED** [review]

**Review by nada-deriv — APPROVED** [review]

File: packages/hooks/src/__tests__/useP2PAdvertInfo.spec.tsx [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+
+        const wrapper = ({ children }: { children: JSX.Element }) => (
+            <APIProvider>
+                <StoreProvider store={mock}>{children}</StoreProvider>
+            </APIProvider>
+        );
+
+        const { result } = renderHook(() => useP2PAdvertInfo('1', { enabled: true }), { wrapper });
+
+        expect(result.current.data?.advertiser_details?.completed_orders_count).toBe(0);
```
Comment (niloofar-deriv) [inline packages/hooks/src/__tests__/useP2PAdvertInfo.spec.tsx]:
```suggestion
const lala = result.current.data?.advertiser_details
expect(lala?.completed_orders_count).toBe(0);
```

Comment (ameerul-deriv) [reply packages/hooks/src/__tests__/useP2PAdvertInfo.spec.tsx]: ah yeah i can make this shorter, and nice name 😉

File: packages/cashier/src/components/error-dialog/error-dialog.tsx [hunk 17]
```diff
@@ -107,6 +107,15 @@ const ErrorDialog = observer(({ className, error = {} }: TErrorDialogProps) => {
...
+                setDetails({
+                    title: localize('Deriv P2P unavailable'),
+                    cancel_button_text: undefined,
+                    confirm_button_text: localize('OK'),
+                    onConfirm: undefined,
+                    message: error_message,
+                    has_close_icon: true,
```
Comment (yashim-deriv) [inline packages/cashier/src/components/error-dialog/error-dialog.tsx]:
```suggestion
                    title: localize('Deriv P2P unavailable'),
                    confirm_button_text: localize('OK'),
                    message: error_message,
                    has_close_icon: true,
```
Removing these props would still make it undefined without the need to explicitly declare them

File: packages/p2p/src/stores/advertiser-page-store.js [hunk 18]
```diff
@@ -107,13 +107,15 @@ export default class AdvertiserPageStore extends BaseStore {
...
-                } else {
-                    const { list } = response.p2p_advert_list;
-
-                    this.setAdverts(list);
-                    this.setHasMoreAdvertsToLoad(list.length >= general_store.list_item_limit);
+                if (response) {
+                    if (response.error) {
+                        this.setErrorMessage(response.error);
+                    } else {
+                        const { list } = response.p2p_advert_list;
```
Comment (yashim-deriv) [inline packages/p2p/src/stores/advertiser-page-store.js:114]:
```suggestion
                        const { list = [] } = response.p2p_advert_list;
```
Can the list (returned by BE) be undefined?

File: packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-modal.tsx [hunk 19]
```diff
@@ -10,46 +10,69 @@ import {
...
+        return type === buy_sell.BUY
+            ? localize(
+                  "Hi! I'd like to exchange {{local_currency}} for {{account_currency}} at {{rate_display}} {{local_currency}} on Deriv P2P.\n\nIf you're interested, check out my ad 👉\n\n{{- advert_url}}\n\nThanks!",
+                  custom_message_values
+              )
+            : localize(
+                  "Hi! I'd like to exchange {{account_currency}} for {{local_currency}} at {{rate_display}} {{local_currency}} on Deriv P2P.\n\nIf you're interested, check out my ad 👉\n\n{{- advert_url}}\n\nThanks!",
+                  custom_message_values
+              );
+    };
```
Comment (yashim-deriv) [inline packages/p2p/src/components/modal-manager/modals/share-my-ads-modal/share-my-ads-modal.tsx]:
```suggestion
    const getCustomMessage = () => {
        if (rate_type === ad_type.FLOAT) {
            return type === buy_sell.BUY
                ? localize(
                      "Hi! I'd like to exchange {{local_currency}} for {{account_currency}} at {{rate_display}}% on Deriv P2P.\n\nIf you're interested, check out my ad 👉\n\n{{- advert_url}}\n\nThanks!",
                      custom_message_values
                  )
                : localize(
                      "Hi! I'd like to exchange {{account_currency}} for {{local_currency}} at {{rate_display}}% on Deriv P2P.\n\nIf you're interested, check out my ad 👉\n\n{{- advert_url}}\n\nThanks!",
                      custom_message_values
                  );
        }
        return type === buy_sell.BUY
            ? localize(
                  "Hi! I'd like to exchange {{local_currency}} for {{account_currency}} at {{rate_display}} {{local_currency}} on Deriv P2P.\n\nIf you're interested, check out my ad 👉\n\n{{- advert_url}}\n\nThanks!",
                  custom_message_values
              )
            : localize(
                  "Hi! I'd like to exchange {{account_currency}} for {{local_currency}} at {{rate_display}} {{local_currency}} on Deriv P2P.\n\nIf you're interested, check out my ad 👉\n\n{{- advert_url}}\n\nThanks!",
                  custom_message_values
              );
    };
```
The base string is the same, only the values differ so a lot of these conditional can be simplified using <Localize /> components via the `values` props.

File: packages/hooks/src/useP2PAdvertInfo.ts [hunk 20]
```diff
@@ -0,0 +1,26 @@
...
+const useP2PAdvertInfo = (id: string, options: TSocketRequestQueryOptions<'p2p_advert_info'>) => {
+    const { data, ...rest } = useFetch('p2p_advert_info', { payload: { id }, options });
+
+    const modified_data = React.useMemo(() => {
+        const p2p_advert_info = data?.p2p_advert_info;
+
+        if (!p2p_advert_info) return undefined;
+
+        return p2p_advert_info;
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (farhan-nurzi-deriv) [inline packages/hooks/src/useP2PAdvertInfo.ts]: Do we need this `eslint` comment 👀

Comment (ameerul-deriv) [reply packages/hooks/src/useP2PAdvertInfo.ts]: ah yeah we can 👍

File: packages/p2p/src/stores/advertiser-page-store.js (hunk 18 above)
Comment (ameerul-deriv) [reply packages/p2p/src/stores/advertiser-page-store.js:114]: just checked, the list wont be undefined, and there will always be an array returned *unless theres an error

Comment (ameerul-deriv) [reply packages/p2p/src/stores/advertiser-page-store.js:114]: just that the above error couldn't recognise response.error, so we needed to check for response first

File: packages/account/package.json [hunk 21]
```diff
@@ -43,7 +43,7 @@
         "mobx-react": "^7.5.1",
         "onfido-sdk-ui": "^11.0.0",
         "prop-types": "^15.7.2",
-        "qrcode.react": "^1.0.0",
+        "qrcode.react": "^3.1.0",
```
Comment (ako-deriv) [inline packages/account/package.json:46]: to apply this on the package-lock file you need to run `npm run bootstrap:dev` from the root level and then commit the package-lock file.

File: packages/cashier/src/components/error-dialog/error-dialog.tsx [hunk 22]
```diff
@@ -13,9 +13,9 @@ type TErrorDialogProps = {
 
 type TSetDetails = {
     title: string;
-    cancel_button_text: undefined | string;
+    cancel_button_text?: undefined | string;
```
Comment (ako-deriv) [inline packages/cashier/src/components/error-dialog/error-dialog.tsx]: since you've made it optional there should be no need for the undefined anymore.
```suggestion
    cancel_button_text?: string;
```

File: packages/cashier/src/components/error-dialog/error-dialog.tsx [hunk 23]
```diff
@@ -13,9 +13,9 @@ type TErrorDialogProps = {
...
     confirm_button_text: undefined | string;
-    onConfirm: undefined | (() => void);
+    onConfirm?: undefined | (() => void);
```
Comment (ako-deriv) [inline packages/cashier/src/components/error-dialog/error-dialog.tsx]:
```suggestion
    onConfirm?: (() => void);
```

File: packages/account/package.json (hunk 21 above)
Comment (amina-deriv) [inline packages/account/package.json:46]: shouldnt we have changes to package-lock.json file?

Comment (amina-deriv) [reply packages/account/package.json:46]: Also this qrcode.react is used in two-factor-authentication (https://app.deriv.com/account/two-factor-authentication). Can we add a sanity test for this in QA checklist

Comment (ameerul-deriv) [reply packages/account/package.json:46]: sure 👍

**Review by ali-hosseini-deriv — APPROVED** [review]

File: packages/p2p/src/components/advertiser-page/advertiser-page.jsx [hunk 24]
```diff
@@ -72,8 +114,20 @@ const AdvertiserPage = () => {
         const disposeCounterpartyAdvertiserIdReaction = reaction(
             () => [general_store.counterparty_advertiser_id, general_store.is_advertiser_info_subscribed],
             () => {
-                // DO NOT REMOVE. This fixes reload on advertiser page routing issue
-                advertiser_page_store.onAdvertiserIdUpdate();
+                if (general_store.counterparty_advertiser_id) {
```
Comment (niloofar-deriv) [inline packages/p2p/src/components/advertiser-page/advertiser-page.jsx]:
```suggestion
                if (counterparty_advertiser_id) {
```

File: packages/p2p/src/components/advertiser-page/advertiser-page.jsx [hunk 25]
```diff
@@ -158,6 +212,7 @@ const AdvertiserPage = () => {
                         if (general_store.active_index === general_store.path.my_profile)
                             my_profile_store.setActiveTab(my_profile_tabs.MY_COUNTERPARTIES);
                         history.push(general_store.active_tab_route);
+                        general_store.setCounterpartyAdvertiserId(null);
```
Comment (niloofar-deriv) [inline packages/p2p/src/components/advertiser-page/advertiser-page.jsx]:
```suggestion
                        setCounterpartyAdvertiserId(null);
```

File: packages/p2p/src/components/advertiser-page/advertiser-page.jsx [hunk 26]
```diff
@@ -20,10 +21,12 @@ import BlockUserOverlay from './block-user/block-user-overlay';
 import classNames from 'classnames';
 import { OnlineStatusIcon, OnlineStatusLabel } from 'Components/online-status';
 import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
+import { buy_sell } from 'Constants/buy-sell';
 import './advertiser-page.scss';
 
 const AdvertiserPage = () => {
     const { advertiser_page_store, buy_sell_store, general_store, my_profile_store } = useStores();
+    const { counterparty_advert_id, is_advertiser, is_barred } = general_store;
```
Comment (niloofar-deriv) [inline packages/p2p/src/components/advertiser-page/advertiser-page.jsx]:
```suggestion
    const { counterparty_advert_id, is_advertiser, is_barred, setCounterpartyAdvertiserId } = general_store;
```

**Review by niloofar-deriv — APPROVED** [review]

**Review by amina-deriv — APPROVED** [review]

**Review by farrah-deriv — APPROVED** [review]

**Review by likhith-deriv — APPROVED** [review]
